addons/repair/XEH_postInit.sqf

```diff
@@ -18,3 +20,65 @@
         _object allowDamage false;
     };
 }] call CBA_fnc_addEventHandler;
+
+
+// Class EHs
+
+["All", "InitPost", {
+    params ["_vehicle"];
+    if (!local _vehicle) exitWith {};
+
+    if (getNumber (configFile >> "CfgVehicles" >> typeOf _vehicle >> "transportRepair") != 0) then {
+        _vehicle setRepairCargo 0;
+    };
+}, true, [], true] call CBA_fnc_addClassEventHandler;
+
+["ACE_RepairItem_Base", "killed", {
+    params ["_object"];
+
+    [{deleteVehicle _this}, _object, 5] call CBA_fnc_waitAndExecute;
+}, true, [], true] call CBA_fnc_addClassEventHandler;
+
+// load tracks and wheels
+if (isServer) then {
+    private _fnc_addSpareItems = {
+        if (!GVAR(addSpareParts)) exitWith {};
+        params ["_vehicle"];
+
+        private _spareTracks = _vehicle getVariable QGVAR(editorLoadedTracks);
+        if (isNil "_spareTracks") then {
+            _spareTracks = [0, 1] select (_vehicle isKindOf "Tank"); // must match eden attribute default
+        };
+        if (_spareTracks > 0) then {
+            [_vehicle, _spareTracks, "ACE_Track"] call FUNC(addSpareParts);
+        };
+
+        private _spareWheels = _vehicle getVariable QGVAR(editorLoadedWheels);
+        if (isNil "_spareWheels") then {
+            _spareWheels = [0, 1] select (_vehicle isKindOf "Car"); // must match eden attribute default
+        };
+        if (_spareWheels > 0) then {
+            [_vehicle, _spareWheels, "ACE_Wheel"] call FUNC(addSpareParts);
+        };
+    };
+
+    ["Tank", "initPost", _fnc_addSpareItems, true, [], true] call CBA_fnc_addClassEventHandler;
+    ["Car", "initPost", _fnc_addSpareItems, true, [], true] call CBA_fnc_addClassEventHandler;
+};
+
+
+// Item replacement EH
+
+["loadout", {
+    params ["_unit"];
+
+    private _cfgWeapons = configFile >> "CfgWeapons";
+    private _toolKits = (items _unit) select {
```

Comment:
VERY bad. `(items _unit)` might be several hundreds of items that we don't care about, and we do a config lookup for EACH of them, EVERY time the players loadout changes.

Was this done in old code or is this a new feature?
Alternatively use uniqueUnitItems if we really need to do this. But I really dislike this happening on every loadout change, how is medical doing item replacement?

https://github.com/acemod/ACE3/blob/44050df98b00e579e5b5a79c0d76d4d1138b4baa/addons/medical_treatment/functions/fnc_checkItems.sqf

Seems like medical is also doing it on every loadout change.. oof.
uniqueUnitItems should be good enough I guess. But if we find items we need to combine with common,getCountOfItem to get the number to replace.
Also needs a setting to disable this.

Reply:
We could (in general, not just this) implement a "delayed item replacement" system.
When a loadout EH is fired we just remember it in a waitAndExecute, if another loadout EH fires we reschedule that waitAndExecute and bump up it's time.

That way if you do a spam loadout change (like grabbing a hundred bandages from a box) it won't be affected. And this item replacement is not a time critical thing, no issue making it happen 5/10 seconds late, same for medical.

Just make a global "delayedItemReplacement" EH that we can then use for that.